### PR TITLE
NIFI-7475 Initial commit for JWT bundle

### PR DESCRIPTION
**Pull request by dml872**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

NIFI-7475 Processor bundle to JWT sign/unsign JSON 

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [X] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [X] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on both JDK 8 and JDK 11?
- [X] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [X] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-jwt-bundle/pom.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    </modules>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.apache.nifi</groupId>
+                <artifactId>nifi-jwt-processors</artifactId>
+                <version>1.12.0-SNAPSHOT</version>
+            </dependency>
+            <!-- <dependency>
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-jwt-bundle/pom.xml:40]: Can we remove completely remove this block instead of commenting?

**Comment by pvillard31** [comment]
Hey @dml872 - thanks a lot for this contribution. I was thinking about it (as I need something very similar at the moment) and I was wondering if it really makes sense to do that based on the flow file content.

My thought process is the following: a user is making a request that is received through a ListenHttp or HandleHttpRequest processors. The HTTP request contains the `Authorization: Bearer <JWT>` header. I would then use the `UnsignJWT` processor to validate the JWT and retrieve the claims from the JWT payload. In my mind the `UnsignJWT` processor would have a property saying in which flow file attribute we have the `Bearer <JWT>` data, would unsign it, and would update the flow file by adding attributes to it with the claims contained in the JWT. If the JWT cannot be unsigned, the flow file would go to a failure relationship.

Does it make sense? Do you have another use case in mind?

**Comment by dml872** [comment]
Thanks @pvillard31 that makes a lot of sense. The use case I had in mind is that the JWT would be in the body of the flow file but I can see more generally that one would expect the JWT in an attribute. 

I can either have 2 different `UnsignJWT` processors, one for the JWT in the body and one for the JWT in a specified attribute. Alternatively I could define another property with options to unsign the body or an attribute?

Which of these would be best, or an alternative?

**Comment by pvillard31** [comment]
I'd suggest adding an optional property where a user can specify in which attribute of the incoming flow file would be the JWT. And the processor would write ``jwt.xxx``` attributes with the claims of the JWT. In such a case the processor would not touch the flow file content and would only work on attributes. If the property is not set by the user, then the processor would expect the JWT to be the flow file content.

File: nifi-nar-bundles/nifi-jwt-bundle/pom.xml (hunk 1 above)
Comment (dml872) [reply nifi-nar-bundles/nifi-jwt-bundle/pom.xml]: d'oh, done!

**Comment by dml872** [comment]
Sorry, got into a bit of a pickle my end but should all be fine now... I added the JWT attribute property to the `UnsignJWT` processor as discussed and the claims are written to `jwt.XXX` attributes. If the property is blank then the JWT is read from the content and claims written to the content still.

**Comment by pvillard31** [comment]
Awesome, thanks @dml872 - will definitely give it a try over the WE.

**Comment by pvillard31** [comment]
Hey @dml872 - sorry for the late reply, I got busy with other things... Can you rebase your pull request against main, squash your commits into a single one and change the references for the NiFi version (it's now `1.13.0-SNAPSHOT`)?

**Comment by dml872** [comment]
Closing this as new PR raised - https://github.com/apache/nifi/pull/4495

**Pull request closed without merging**